### PR TITLE
Implement AccessControlException.

### PR DESCRIPTION
**Pull request by andreaTP**
Started looking at Pickling (https://github.com/andreaTP/pickling) .
This is just the first linking error I've got....

Any help on this is much appreciated!

### CODE REVIEW

File: javalib/src/main/scala/java/security/Guard.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package java.security
+
+trait Guard {
+  def checkGuard(o: Any): Unit
```
Comment (nicolasstucki) [inline javalib/src/main/scala/java/security/Guard.scala]: This should probably be `def checkGuard(o: AnyRef): Unit`

File: javalib/src/main/scala/java/security/Permission.scala [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+abstract class Permission(name: String) extends Guard with Serializable {
+  //def checkGuard(a: Any): Unit
+  def implies(p: Permission): Boolean
+  def equals(obj: Any): Boolean
+  def hashCode(): Int
+  //final def getName(): String
+  def getActions(): String
+  //def newPermissionCollection(): PermissionCollection
+  //def toString(): String
```
Comment (nicolasstucki) [inline javalib/src/main/scala/java/security/Permission.scala]: Why did you leave some of the abstract members commented out?

File: javalib/src/main/scala/java/security/Permission.scala [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package java.security
+
+abstract class Permission(name: String) extends Guard with Serializable {
+  //def checkGuard(a: Any): Unit
```
Comment (nicolasstucki) [inline javalib/src/main/scala/java/security/Permission.scala]: `checkGuard` is supposed to be implemented in `Permission`.

File: javalib/src/main/scala/java/security/Permission.scala (hunk 2 above)
Comment (nicolasstucki) [inline javalib/src/main/scala/java/security/Permission.scala]: The `toString` has a simple implementation based on the other members.

File: javalib/src/main/scala/java/security/Throwables.scala [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package java.security
+
+class SecurityException(message: String, cause: Throwable)
```
Comment (nicolasstucki) [inline javalib/src/main/scala/java/security/Throwables.scala]: `SecurityException` should be in `java.lang`.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/ThrowablesTest.scala [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib.security
+
+import org.junit.Test
+
+class ThrowablesTest {
+  @Test def should_define_all_java_security_Errors_and_Exceptions(): Unit = {
+    import java.security._
+    new SecurityException("", new Exception())
```
Comment (nicolasstucki) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/ThrowablesTest.scala]: `SecurityException` should go in `org.scalajs.testsuite.javalib.lang.ThrowablesTest`.

**Comment by nicolasstucki** [comment]
That is all.

File: javalib/src/main/scala/java/security/Guard.scala (hunk 1 above)
Comment (andreaTP) [inline javalib/src/main/scala/java/security/Guard.scala]: We usually use ```Any```
https://docs.oracle.com/javase/7/docs/api/java/util/AbstractCollection.html#contains(java.lang.Object)
https://github.com/scala-js/scala-js/blob/master/javalib/src/main/scala/java/util/AbstractCollection.scala#L14

File: javalib/src/main/scala/java/security/Permission.scala (hunk 2 above)
Comment (andreaTP) [inline javalib/src/main/scala/java/security/Permission.scala]: abstract members are all uncommented, I hope :-)
Ok on "toString"

File: javalib/src/main/scala/java/security/Permission.scala (hunk 3 above)
Comment (andreaTP) [inline javalib/src/main/scala/java/security/Permission.scala]: I just need the interface to let the Exception available without linking errors, I have no feasible way to implement it ...

File: javalib/src/main/scala/java/security/Throwables.scala (hunk 4 above)
Comment (andreaTP) [inline javalib/src/main/scala/java/security/Throwables.scala]: You are right!!

File: javalib/src/main/scala/java/security/Guard.scala (hunk 1 above)
Comment (nicolasstucki) [inline javalib/src/main/scala/java/security/Guard.scala]: Ok

File: javalib/src/main/scala/java/security/Permission.scala (hunk 3 above)
Comment (nicolasstucki) [inline javalib/src/main/scala/java/security/Permission.scala]: OK

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/ThrowablesTest.scala (hunk 5 above)
Comment (andreaTP) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/ThrowablesTest.scala]: ok

**Comment by andreaTP** [comment]
ok, please review this.

**Comment by nicolasstucki** [comment]
Jenkins, test this please

**Comment by nicolasstucki** [comment]
LGTM

**Pull request merged**